### ISSUE TITLE
Compat Mode: Checking for "sessionStorage" reference crashes the app when chrome has "block third-party cookies" enabled

### ISSUE THREAD

**Issue by rosostolato** [firebase/firebase-js-sdk#5913]
The same as https://github.com/firebase/firebase-js-sdk/issues/5618 but it's still broken for compat mode.

If customers enable the "block third-party cookies" feature under chrome settings, Chrome will throw an exception when you try to access the `sessionStorage` reference in `window`. To solve that, we should add a `try/catch` for all lines that reference `sessionStorage` and not only check if it's null.

https://github.com/firebase/firebase-js-sdk/blob/cdada6c68f9740d13dd6674bcb658e28e68253b6/packages/auth-compat/src/persistence.ts#L104

```
DOMException: Failed to read the 'sessionStorage' property from 'Window': Access is denied for this document.
Unhandled Promise rejection: The user denied permission to access the database. ; Zone: <root> ; Task: Promise.then ; Value: The user denied permission to access the database.
```

**Comment by rosostolato** [comment]
Same as https://github.com/firebase/firebase-js-sdk/issues/5618

**Comment by hsubox76** [comment]
Seems to be used in Auth and Database

**Comment by sam-gc** [comment]
That PR I just linked should fix it in Auth, but I'll make sure to leave this open for database as well. @rosostolato I only see the issue when all cookies are disabled, not just third-party cookies. Which version of Chrome are you on?

**Comment by rosostolato** [comment]
@sam-gc I don't remember the version of Chrome I was using but I have just tested it now with the latest one and it seems it's not crashing anymore. I still see the database access error, but it's not breaking the app.

**Comment by sam-gc** [comment]
@rosostolato no problem, like I said it still happens for me when all cookies are disabled so there's at least an easy way to reproduce it :+1:

**Comment by sam-gc** [comment]
I'm going to remove the `api:auth` tag for now; the fix for auth should be in the next release or the one after. We will keep this issue open for database

**Comment by wisonye** [comment]
> That PR I just linked should fix it in Auth, but I'll make sure to leave this open for database as well. @rosostolato I only see the issue when all cookies are disabled, not just third-party cookies. Which version of Chrome are you on?

Hi there, actually, I still got the same error after installing the `latest firebase`(^9.6.8), and I've already confirmed that install "@firebase/auth-compat" on "0.2.9" version (search in `package-lock.json`).

This error still will happen when running the Firebase app in an iframe (when `window.sessionStoarge` is called inside the firebase source code. I think `_getSelfWindow()?.sessionStorage` ???).

But finally, I realize that we don't need to save the token across sessions (or cross-browser tabs), so I changed to `inMemoryPersistence` which can save my life, as it seems there is NO WAY to avoid that error when the `Block third-party` flag is enabled, plz correct me if I'm wrong:)
